File: lib/csharp/microsoft.bot.builder.solutions.skills/Microsoft.Bot.Builder.Solutions.Skills/Models/Manifest/SkillManifest.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using Microsoft.Bot.Builder.Skills;
using Newtonsoft.Json;

namespace Microsoft.Bot.Builder.Solutions.Skills.Models.Manifest
{
    /// <summary>
    /// The SkillManifest class models the Skill Manifest which is used to express the capabilities
    /// of a skill and used to drive Skill configuration and orchestration.
    /// </summary>
    public class SkillManifest : SkillOptions
    {
        [JsonProperty(PropertyName = "description")]
        public string Description { get; set; }

        [JsonProperty(PropertyName = "iconUrl")]
        public Uri IconUrl { get; set; }

        [JsonProperty(PropertyName = "authenticationConnections")]
        public List<AuthenticationConnection> AuthenticationConnections { get; } = new List<AuthenticationConnection>();

        [JsonProperty(PropertyName = "actions")]
        public List<Action> Actions { get; } = new List<Action>();
    }
}

File: sdk/typescript/libraries/bot-solutions/src/skills/models/manifest/skillManifest.ts
/**
 * Copyright(c) Microsoft Corporation.All rights reserved.
 * Licensed under the MIT License.
 */

import { IAction, IAuthenticationConnection } from './';

/**
 * The SkillManifest class models the Skill Manifest which is used to express the capabilities
 * of a skill and used to drive Skill configuration and orchestration.
 */
export interface ISkillManifest {
    id: string;
    msaAppId: string;
    name: string;
    endpoint: string;
    description: string;
    iconUrl: string;
    authenticationConnections: IAuthenticationConnection[];
    actions: IAction[];
}
